### DISCUSSION TITLE
[Question] How do I create a custom mesh from a list/array of vertices?

### DISCUSSION THREAD

**Discussion by Mahmoud1205** [jrouwe/JoltPhysics#1112] [Q&A] [answered]
This is my Vertex struct
```cpp
struct Vertex {
    glm::vec3 Position;
    glm::vec3 Normal;
    glm::vec2 TexCoords
}
```
I have a list of indices too

**Comment by jrouwe** [comment, marked as answer]
Here's an example:

https://github.com/jrouwe/JoltPhysics/blob/6370b0f0873c612dd2c8f0b3ea3d085d0e6a4bdb/Samples/Tests/Character/CharacterBaseTest.cpp#L413-L455

**Reply by Mahmoud1205** [reply]
Thank you, does the engine automatically optimize these (as in decreasing amount of vertices, if that even matters) or should I decrease the vertices of the collision body myself?

**Reply by jrouwe** [reply]
It only removes degenerate triangles, it doesn't do any further simplifications. Use another mesh simplification library if you want to simplify the mesh.
